.github/workflows/ci.yml

```diff
@@ -5,28 +5,297 @@ on:
+      - name: Compile dependencies
+        if: steps.deps-cache.outputs.cache-hit != 'true'
+        run: mix deps.compile
+      
+      - name: Run tests
+        run: |
+          mix test
+      
+      - name: Check formatting
+        run: |
+          mix format --check-formatted
+      
+      - name: Compile with warnings as errors
+        run: |
+          mix compile --warnings-as-errors --force
+
+      - name: Dialyzer
+        run: |
+          mix dialyzer --force-check
+
+      - name: Credo
+        run: |
+          mix credo
+
+      - name: Sobelow
+        run: |
+          mix sobelow --config
+
+      - name: Deps Unused
+        run: |
+          mix deps.unlock --check-unused
+
+      - name: Deps Audit 
+        continue-on-error: true
+        run: |
+          mix deps.audit
+      
+      - name: Gettext Check
+        run: |
+          mix gettext.extract --check-up-to-date
+
+  build-testing:
+    name: "Build testing"
+    runs-on: ubuntu-latest
 
-      - name: Run Tests
-        run: mix test
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_TESTING }}
+
+      - name: Build container
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=registry,ref=${{ env.REGISTRY }}/${{ env.IMAGE_NAME_TESTING }}:buildcache
+          cache-to: type=registry,ref=${{ env.REGISTRY }}/${{ env.IMAGE_NAME_TESTING }}:buildcache,mode=max
+          target: builder
+          build-args: |
+            MIX_ENV=test
+    outputs:
+      image: ${{ steps.meta.outputs.tags }}
+
+  lint-demo:
+    needs: build-testing
+    name: "Lint (Demo)"
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ needs.build-testing.outputs.image }}
+  
+    steps:
+      - name: lint:mix
+        working-directory: /opt/app/demo
+        run: |
+          yarn lint:mix
+          
+      - name: Add tar that supports --posix option to be used by cache action
+        run: |
+          apk add --no-cache tar
+
+      - name: Restore the plts cache
+        uses: actions/cache@v4
+        id: plts-cache
+        with:
+          path: /opt/app/demo/priv/plts
+          key: ${{ runner.os }}-demo-plts-mixlockhash-${{ hashFiles(format('{0}{1}', github.workspace, '/opt/app/demo/mix.lock')) }}
```

Comment:
This `key` does not seem to work properly, because the cache is always saved on the primary key `Linux-demo-plts-mixlockhash-` (see [this](https://github.com/naymspace/backpex/actions/runs/7931127493/job/21654741621) pipeline). The hash of the `mix.lock` does not get attached. 

When do we even want to invalidate the demo plts cache?

Reply:
We should remove Dialyzer. See also internal discussion.